start.py

```diff
@@ -35,20 +35,25 @@
 def main(args):
     kerberos_volume_dir = os.path.expanduser(args.kerberos_config_directory or args.clusterdock_config_directory)
 
-    #kdc_image = '{}/{}/topology_http_kerberos:kdc'.format(args.registry,
-    #                                                      args.namespace or DEFAULT_NAMESPACE)
+    # kerberos node.
     kdc_image = '{}/clusterdock/topology_nodebase_kerberos:centos6.8'.format(args.registry)
     kdc_hostname = args.kdc_node[0]
     kdc_node = Node(hostname=kdc_hostname, group='kdc', image=kdc_image,
                     volumes=[{kerberos_volume_dir: KERBEROS_VOLUME_DIR}])
 
+    # webserver node. this is the reverse proxy that exposes the URLs
     webserver_image = '{}/{}/topology_http_kerberos:webserver'.format(args.registry,
                                                                       args.namespace or DEFAULT_NAMESPACE)
     webserver_hostname = args.webserver_node[0]
     webserver_node = Node(hostname=webserver_hostname, group='webserver', image=webserver_image,
-                          volumes=[{kerberos_volume_dir: KERBEROS_VOLUME_DIR}], ports={80: 80})
+                          volumes=[{kerberos_volume_dir: KERBEROS_VOLUME_DIR}], ports={80: 80, 443: 443})
 
-    cluster = Cluster(kdc_node, webserver_node)
+    # service node. the actual service (in our case, pretenders, which allows us to create mock http urls)
+    service_hostname = args.service_node[0]
+    service_node = Node(hostname=service_hostname, group='service', image='pretenders/pretenders:1.4',
+                        ports={8000: 8000})
```

Comment:
So one of the things about clusterdock has always been that each "node" acts like a real cluster node (e.g. you can SSH into it via `clusterdock ssh node-1.cluster`). I imagine the pretenders Docker image wouldn't allow for this, so should this be concerning to us?

Reply:
Yes, this is a bit different from the standard clusterdock node. One of the reasons, I added this in was because without it the topology was kind of incomplete because it depended on another component. It wasn't very useful by itself. I could look into seeing if I create our node image for pretenders that is like a true clusterdock 'Node'. I would prefer to do that as a separate change though.